From mboxrd@z Thu Jan  1 00:00:00 1970
Return-Path: <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 2CFD8D30CC0
	for <linux-rockchip@archiver.kernel.org>; Tue, 13 Jan 2026 20:00:02 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-ID:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=lFt3bx8geZ4Td2MZPi9tmRWnIR+Xvx43o2sY3G/zoTI=; b=u8ASVyfzSR35Y2
	oSieWbwZ/nxxOkv0uFEjQ1r3Miw41FNldb8mt2Gao2ZIL+pOQy6XFMX68TyutrKpCwII/Wu2e8JKy
	HyGSn9BuapDHDqUmfHnLnlpb/X1iae7hQHyUqkrcutlPgKdN+KQ+uGS6bZXjkN66/QAf0gWswY3bq
	/Jt7M33hIiVAnaQXjua4by+QvSPFON1aeJw9M+T/pdsVDq3svpmcHOK2Ov6G4OA/mhkY5bNBDwjhS
	eIIVaStD5iK6SyhljcTwcU9YsrPQxlUm/NVPWRHfEqSwln6s5VNG/YySOXrLlf3LKVtaBmKdYv4nS
	i/AVbSnHqxI+GvvavNcg==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.98.2 #2 (Red Hat Linux))
	id 1vfkYS-00000007eZ9-3CKQ;
	Tue, 13 Jan 2026 19:59:52 +0000
Received: from mail-ot1-x332.google.com ([2607:f8b0:4864:20::332])
	by bombadil.infradead.org with esmtps (Exim 4.98.2 #2 (Red Hat Linux))
	id 1vfkYP-00000007eXf-2WSQ
	for linux-rockchip@lists.infradead.org;
	Tue, 13 Jan 2026 19:59:50 +0000
Received: by mail-ot1-x332.google.com with SMTP id 46e09a7af769-7c7aee74dceso3447684a34.2
        for <linux-rockchip@lists.infradead.org>; Tue, 13 Jan 2026 11:59:49 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20230601; t=1768334388; x=1768939188; darn=lists.infradead.org;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:from:to:cc:subject:date
         :message-id:reply-to;
        bh=dJ3Ahf49hlWnxKtsDsKhHCo3RwD4rUKnAOeXD4F8idM=;
        b=ky/LWf17M/HG/gYgBeKdD2sxk8cT8p/nR7iYsRIBHzYD1OSX65ivwTwiPhvRt2p+bh
         b7r5OeKn629+lo4NJ69UOn/WIFK8cBNWt0rzHpYzqSF3P8RduOfX086pMx59pwzp1nrm
         PkOgsCmUU/veRMMT/UpA7McbHlS5KK2QdZStGiShcH9Rh9GW2IoQYb7RhWQu8zRdEOtP
         kBtDldMdAd7cIk+otzo0mE/r82/Xd80QV+YmN9cqQ7hc7kLbvbtw3fY00SIBuyse0O1U
         ZA6Y22nKo/NNAhSFT7Jg2Ji4uRD9t3Z3/i1SnFldEgUb7pMiUQEBzqF/fxVDEAnXfM/r
         TGJA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1768334388; x=1768939188;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-gg:x-gm-message-state:from
         :to:cc:subject:date:message-id:reply-to;
        bh=dJ3Ahf49hlWnxKtsDsKhHCo3RwD4rUKnAOeXD4F8idM=;
        b=HKMEAQPODga8VGwsnv6Mir6waxFXkY7tcqQEZjdK1w2+JZFIn/k77vE8t8GGOO7Z2F
         Wg18z0L0aVQJa+iF1WVzBIjcILKAbNhlPWs6lS/vANePPpgZV4ddu0fXy2QdP+0noKiO
         Q5gtqXGhw12FD0Sdt4XwoQferxZMWum3X4bwkr9frEMsjnTyaDnqRsxMIeQAKpbGjjOX
         CGnlBSF1T1wu1QY/ocvodp0cXxHYAA7db0Ub7sl+keclIbDeboHG34+3ZHMiMc+cSRgY
         HZzJkrg3bdfN5WnFiyde09v8yUksXvf8i2WWvs4QfdRvX7yFWCAlNUWDdzyXahCEnoOq
         wRPg==
X-Gm-Message-State: AOJu0Yzo5CvXCcYS4XcM6cNM+JDmUZFH1cyYUpoXEucTzKnO4SYf545H
	q8JUmciUH6mPTBpNuzPh3Gz0p1opPo4Fbj3lRSxxfwv48B6YHQwabZA/lLu4Aw==
X-Gm-Gg: AY/fxX5H0ywOezWzDz0nbCZZFei8gBk1QKC/hVY7kKYdlYiIrWx1ifGv+5oXqFXkGqF
	3w26uHs6G1BwDLt2T1VML3XUhpjub/GmVnQXU547Rj9hO8FcRcUhlt23RHNx4nljc9VRZLnVN7S
	aftSB01vj6O+D55N/Wwi3QdqWZdj8FIspcLAKEceNrsIo9EMIPipyI7SDHQFoqeW87njp8RWU51
	bMsdUWMwPqkwZ4LZY4fjtiW2OG/jUq1YenC4FnZ2xdkCsquf3XI3AOqx22ix8FRwE4E/GCaT1Xn
	u1xwyoTbXGtPvoJOIorMc07f6StAIEZiSa83FvSG8IMPZZZKYuclMA7O1XnJpS67rsDTjpBz/MN
	ObZBfe8+1D/15yVZTfABr8XZIMdg2dT7IJvvZXWePZNLnLPmZ5e6+GpI5AcSHBe0szpVkXMGhTe
	lseM6BW6uS
X-Received: by 2002:a05:6830:6408:b0:7bc:31fe:57a9 with SMTP id 46e09a7af769-7cfc8b3b8f6mr195237a34.31.1768334388271;
        Tue, 13 Jan 2026 11:59:48 -0800 (PST)
Received: from localhost.localdomain ([2600:1700:fb0:1bc0::54])
        by smtp.gmail.com with ESMTPSA id 46e09a7af769-7ce47801d4dsm16221119a34.5.2026.01.13.11.59.47
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 13 Jan 2026 11:59:47 -0800 (PST)
From: Chris Morgan <macroalpha82@gmail.com>
To: linux-rockchip@lists.infradead.org
Cc: dri-devel@lists.freedesktop.org,
	devicetree@vger.kernel.org,
	dmitry.torokhov@gmail.com,
	simona@ffwll.ch,
	airlied@gmail.com,
	tzimmermann@suse.de,
	mripard@kernel.org,
	maarten.lankhorst@linux.intel.com,
	jesszhan0024@gmail.com,
	neil.armstrong@linaro.org,
	jagan@edgeble.ai,
	heiko@sntech.de,
	conor+dt@kernel.org,
	krzk+dt@kernel.org,
	robh@kernel.org,
	aweinzerl13@yahoo.com,
	Chris Morgan <macromorgan@hotmail.com>
Subject: [PATCH 1/6] drm: panel: jd9365da: Use gpiod_set_value_cansleep()
Date: Tue, 13 Jan 2026 13:57:16 -0600
Message-ID: <20260113195721.151205-2-macroalpha82@gmail.com>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20260113195721.151205-1-macroalpha82@gmail.com>
References: <20260113195721.151205-1-macroalpha82@gmail.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20260113_115949_648975_BAB4F057 
X-CRM114-Status: GOOD (  10.33  )
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

From: Chris Morgan <macromorgan@hotmail.com>

Change instances of gpiod_set_value() to gpiod_set_value_cansleep().
Uses of gpiod_set_value() generates warnings when used in instances
where desc->gdev->can_sleep is true.

Signed-off-by: Chris Morgan <macromorgan@hotmail.com>
---
 drivers/gpu/drm/panel/panel-jadard-jd9365da-h3.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/panel/panel-jadard-jd9365da-h3.c b/drivers/gpu/drm/panel/panel-jadard-jd9365da-h3.c
index aa05316dc57b..063ed9646d53 100644
--- a/drivers/gpu/drm/panel/panel-jadard-jd9365da-h3.c
+++ b/drivers/gpu/drm/panel/panel-jadard-jd9365da-h3.c
@@ -109,13 +109,13 @@ static int jadard_prepare(struct drm_panel *panel)
 	if (jadard->desc->lp11_to_reset_delay_ms)
 		msleep(jadard->desc->lp11_to_reset_delay_ms);
 
-	gpiod_set_value(jadard->reset, 0);
+	gpiod_set_value_cansleep(jadard->reset, 0);
 	msleep(5);
 
-	gpiod_set_value(jadard->reset, 1);
+	gpiod_set_value_cansleep(jadard->reset, 1);
 	msleep(10);
 
-	gpiod_set_value(jadard->reset, 0);
+	gpiod_set_value_cansleep(jadard->reset, 0);
 	msleep(130);
 
 	ret = jadard->desc->init(jadard);
@@ -129,11 +129,11 @@ static int jadard_unprepare(struct drm_panel *panel)
 {
 	struct jadard *jadard = panel_to_jadard(panel);
 
-	gpiod_set_value(jadard->reset, 0);
+	gpiod_set_value_cansleep(jadard->reset, 0);
 	msleep(120);
 
 	if (jadard->desc->reset_before_power_off_vcioo) {
-		gpiod_set_value(jadard->reset, 1);
+		gpiod_set_value_cansleep(jadard->reset, 1);
 
 		usleep_range(1000, 2000);
 	}
-- 
2.43.0


_______________________________________________
Linux-rockchip mailing list
Linux-rockchip@lists.infradead.org
http://lists.infradead.org/mailman/listinfo/linux-rockchip

